File: java-home-inventory-core/src/br/com/sailboat/homeinventory/core/repository/ProductRepository.java
package br.com.sailboat.homeinventory.core.repository;

import java.util.List;

import br.com.sailboat.homeinventory.core.Filter;
import br.com.sailboat.homeinventory.core.entity.Product;

public interface ProductRepository {
    void save(Product product);
    void remove(Product product);
    List<Product> getAll(Filter filter);
    Product findById(Long productId);
}

File: android-home-inventory-app/app/src/main/kotlin/br/com/sailboat/homeinventory/domain/repository/ProductRepository.kt
package br.com.sailboat.homeinventory.domain.repository

import br.com.sailboat.homeinventory.domain.Either
import br.com.sailboat.homeinventory.domain.entity.Product
import br.com.sailboat.homeinventory.domain.failure.Failure


interface ProductRepository {

    fun getProduct(productId: Long): Product
    fun getProducts(): Either<Failure, List<Product>>
    fun getProductList(): List<Product>
    fun insert(product: Product)
    fun update(product: Product)
    fun remove(product: Product)

}